tests/unit/main_html5/wrapper-event-tests.js

```diff
@@ -450,7 +450,114 @@ describe('main_html5 wrapper tests', function () {
+    var isLive = true;
+    OO.isSafari = true;
+    element.textTracks = [
+      { language: "", label: "", kind: "subtitles" },
+      { language: "", label: "", kind: "subtitles" },
+      { language: "", label: "", kind: "subtitles" }
+    ];
+    wrapper.setVideoUrl("url", OO.VIDEO.ENCODING.HLS, isLive);
+    wrapper.setClosedCaptions("en", closedCaptions, { mode: "hidden" });
+    $(element).triggerHandler("playing");
+    expect(vtc.notifyParameters).to.eql([vtc.interface.EVENTS.CAPTIONS_FOUND_ON_PLAYING, {
+      languages: ['en', 'CC1', 'CC2', 'CC3'],
+      locale: {
+        en: 'English',
+        CC1: 'Captions (CC1)',
+        CC2: 'Captions (CC2)',
+        CC3: 'Captions (CC3)'
+      }
+    }]);
+  });
+
+  it('should notify CAPTIONS_FOUND_ON_PLAYING with multiple in-manifest/in-stream captions using label and language metadata when available', function() {
+    var isLive = true;
+    OO.isSafari = true;
+    element.textTracks = [
+      { language: "", label: "", kind: "subtitles" },
+      { language: "es", label: "", kind: "subtitles" },
+      { language: "de", label: "German", kind: "subtitles" }
+    ];
+    wrapper.setVideoUrl("url", OO.VIDEO.ENCODING.HLS, isLive);
+    wrapper.setClosedCaptions("en", closedCaptions, { mode: "hidden" });
+    $(element).triggerHandler("playing");
+    expect(vtc.notifyParameters).to.eql([vtc.interface.EVENTS.CAPTIONS_FOUND_ON_PLAYING, {
+      languages: ['en', 'CC1', 'CC2', 'CC3'],
+      locale: {
+        en: 'English',
+        CC1: 'Captions (CC1)',
+        CC2: 'es',
+        CC3: 'German'
+      }
+    }]);
+  });
+
+  it('should reset in-manifest/in-stream track ids when a new source is set', function() {
+    var isLive = true;
+    OO.isSafari = true;
+    element.textTracks = [
+      { language: "", label: "", kind: "subtitles" },
+      { language: "", label: "", kind: "subtitles" },
+    ];
+    wrapper.setVideoUrl("url1", OO.VIDEO.ENCODING.HLS, isLive);
+    $(element).triggerHandler("playing");
+    expect(vtc.notifyParameters).to.eql([vtc.interface.EVENTS.CAPTIONS_FOUND_ON_PLAYING, {
+      languages: ['CC1', 'CC2'],
+      locale: {
+        CC1: 'Captions (CC1)',
+        CC2: 'Captions (CC2)',
+      }
+    }]);
+    // Tracks should start from 1 once more instead of CC3, CC4...
+    element.textTracks = [
+      { language: "", label: "", kind: "subtitles" },
+      { language: "", label: "", kind: "subtitles" },
+    ];
+    wrapper.setVideoUrl("url2", OO.VIDEO.ENCODING.HLS, isLive);
+    $(element).triggerHandler("playing");
+    expect(vtc.notifyParameters).to.eql([vtc.interface.EVENTS.CAPTIONS_FOUND_ON_PLAYING, {
+      languages: ['CC1', 'CC2'],
+      locale: {
+        CC1: 'Captions (CC1)',
+        CC2: 'Captions (CC2)',
+      }
+    }]);
+  });
+
+  it('should NOT readd manually added tracks to available captions when in-manifest/in-stream tracks are checked', function() {
```

Comment:
extra d in readd

Reply:
It's actually meant to be `re-add`, will fix